lib/utils.js

```diff
@@ -95,15 +139,33 @@ const utils = {
         utils.logError('Configuration file missing for this directory, run losant configure to generate this file.');
         return;
       }
-      config.file = file;
+      if (!config.apiUrl) {
+        const url = utils.getApiURL(userConfig);
+        if (url) {
+          config.apiUrl = url;
+          await writeFile(foundFile, yaml.safeDump(userConfig));
+        }
+      }
+      config.file = foundFile;
     }
     return config;
   },
 
   loadConfig: async (file) => {
-    const userConfig = await utils.loadUserConfig(file);
-    const appConfig = await utils.loadApplicationConfig(file);
-    return merge(userConfig, appConfig);
+    let userConfig = await utils.loadUserConfig();
+    const appConfig = await utils.loadApplicationConfig(file, userConfig);
+    if (!appConfig) { return {}; }
+    // backwards compatible in case the current applicationConfig does not have an apiUrl
+    const apiKey = appConfig.apiUrl;
+    userConfig = userConfig[apiKey];
+    if (!userConfig || isEmpty(userConfig)) {
+      utils.logError(`Could not find log in information for API ${apiKey}, please login again.`);
```

Comment:
I think you want two different error messages - one for no user config at all, one for no userConfig for this apiUrl

Reply:
the no user config at all case is handled in the loadUserConfig function and then exits immediately. This is the cannot find the userConfig for this configuration error messages.